spectrum-network/src/network_controller.rs

```diff
@@ -455,6 +586,14 @@ where
                     }
                     continue;
                 }
+                Poll::Ready(Some(PeerManagerOut::NotifyPeerPunished { peer_id, reason })) => {
+                    self.peer_punished(peer_id, reason);
+                    continue;
+                }
+                Poll::Ready(Some(PeerManagerOut::NotifyConnectionLost { peer_id, reason })) => {
+                    self.peer_disconnected(peer_id, reason);
```

Comment:
Why is PeerManager a source of NotifyConnectionLost? Connection loss happens is first observed on PCH and then propagated to NC, why PM notifies NC here?

Reply:
Right, I've removed it. I've moved the call to `self.peer_disconnected(peer_id, reason);` into `NetworkController::inject_connection_closed` instead. This preserves the results of the integration tests.